cpp/frameProcessor/include/FrameProcessorPlugin.h

```diff
@@ -90,45 +91,32 @@ class FrameProcessorPlugin : public IFrameCallback, public OdinData::IVersionedO
    */
   void add_metadata(OdinData::IpcMessage& message, const ParamMetadata& metadata) const
   {
-    std::string str = "metadata/";
-    static std::string& str_static = str;
-    static OdinData::IpcMessage&& message_static = std::move(message);
-
-    str += metadata.path;
-
+    std::string str = get_name() + metadata.path + '/';
```

Comment:
This needs to be nested under "metadata" otherwise it will clash with the actual values.

```suggestion
    std::string str = "metadata" + get_name() + metadata.path + '/';
```

Reply:
Hi Gary,

I looked at one of the sample json structure you shared on issue #407 :
```
{
    "metadata": {
        "ctrl_endpoint": {"type": "str", "access_mode": "rw"},
        "hdf": {
            "file": {
                "first_number": {"type": "bool", "access_mode": "rw"}
            },
            "frames": {"type": "int", "access_mode": "rw", "min": 0}
        },
}
```
I assumed the `"metadata"` part of the string was going to be within the `metadata.path` string.